dali/base/dasds.cpp

```diff
@@ -2778,12 +2787,26 @@ class CNodeSubscriptionManager : public CSimpleInterface, implements INodeSubscr
         CriticalBlock b(lock);
         _notify(&node, state);
     }
-    void notifyDelete(CServerRemoteTree &node)
+    void notifyDelete(CServerRemoteTree *node)
     {
         // shouldn't be here, unless node is in subscribers table
-        CriticalBlock b(lock);
-        _notify(&node, PDS_Deleted);
-        _removeNode(&node, 0);
+        {
+            CriticalBlock b(lock);
+            /* Need to be careful not to release subscribers here (on this thread)
+             * 1) gather subscribers(linked)
+             * 2) remove nodes and lists, so no longer in use by SDS
+             * 3) Hand ownership over to notification mechanism
+             *
+             * Subscribers will be released when notification is done with them.
+             */
+            CNodeSubscriberContainerList *subscriberList = subscriberListByNode.find(node);
+            assertex(subscriberList);
+            IArrayOf<CNodeSubscriberContainer> linkedSubscribers;
```

Comment:
No, it is only to keep them alive after the _removeNode() call, which could otherwise delete.
Then _notify() will consume them.